celery/worker/consumer/consumer.py

```diff
@@ -336,6 +351,15 @@ def on_connection_error_after_connected(self, exc):
         except Exception:  # pylint: disable=broad-except
             pass
 
+        if self.app.conf.worker_cancel_long_running_tasks_on_connection_loss:
+            for request in tuple(active_requests):
+                if request.task.acks_late and not request.acknowledged:
+                    warn(TERMINATING_TASK_ON_RESTART_AFTER_A_CONNECTION_LOSS,
+                         request)
```

Comment:
Should this be using `request.task.id` to ensure we only stringify the ID? Happy for this to remain as is or change as appropriate.

Reply:
No, we need to stringify the request as intended.